File: sorting/insertion/insertion.go
package main

import "fmt"
import "math/rand"

func getRandomNumberList(size int) []int {

	numbers := make([]int, size)

	for idx, _ := range numbers {
		numbers[idx] = rand.Intn(9)
	}

	return numbers
}

func sort(numbers []int) []int {

	for i := 1; i < len(numbers); i++ {

		key := numbers[i]
		j := i - 1

		for j >= 0 && numbers[j] > key {
			numbers[j+1] = numbers[j]
			j--
		}
		numbers[j+1] = key

	}

	return numbers
}

func main() {

	numbers := getRandomNumberList(30)
	fmt.Println("Generated a list of random numbers:", numbers)
	fmt.Println("Sorted list: ", sort(numbers))
}

File: sorting/insertion/insertion.py
import random

def sort(numbers):
    """Sort a list of numbers using `insertion sort` algorithm."""
    return numbers


if __name__ == '__main__':
    numbers = [ random.randint(0, 100) for x in range(0, 100) ]
    _sorted = sort(numbers)
    assert _sorted == sorted(numbers), f"Its not sorted :(\n{_sorted}"